src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -4067,6 +4083,22 @@ def update_enable_advanced_network_observability_in_network_profile(self, mc: Ma
             )
         return mc
 
+    def update_advanced_networking_observability_tls_management(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update advanced network observability tls mangement of network profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        tls_management = self.context.get_advanced_networking_observability_tls_management()
+        if tls_management is not None:
+            if mc.network_profile.advanced_networking is not None:
```

Comment:
What if advance_networking is None? Can't the user enable observability tls management?

Reply:
My understanding is that users should only be able to enable observability tls management when observability enabled is true and that can only be true when advance_networking is not none.

Reply:
OK, then I think maybe it's better to combine update_enable_advanced_network_observability_in_network_profile with update_advanced_networking_observability_tls_management, as both of them are updating the observability sub-profile.

By the way, consider adding some warnings (better do the validation in the getter functions, e.g., get_advanced_networking_observability_tls_management) when users update with only tls_management set and without observability enabled, otherwise the user input will be silently ignored, which will confuse _some_ customers.

Reply:
Updated!